### DISCUSSION TITLE
context rule patterns

### DISCUSSION THREAD

**Discussion by Gopalakrishnan11** [medspacy/medspacy#255] [General]
Hi 
In the context, the rule patterns I have given
```
{
      "category": "POSSIBLE_EXISTENCE",
      "literal": "not helps to cure",
      "pattern": [
        {
          "LOWER": {
            "IN": ["not", "n't"]
          }
        },
        {
          "LOWER": {
            "IN": ["to"]
          }
        },
        {
          "LOWER": {
            "IN": ["cures","curing","cure"]
          }
        },
        {
          "LOWER": {
              "IN":["help","helps"]
          }
        }
      ],
      "direction": "FORWARD"
    }
```
above this to avoid the symptoms being negated. but it still negates the symptoms.
for eg:
`sentence = medication not helps cure anxiety`
For the above sentence, below is the DEP parsing.
![CaptureOct10](https://github.com/medspacy/medspacy/assets/105412331/0f4d86a8-8274-41bf-9663-21f2cb5d10b8)

So, please help me with this issue.

**Comment by burgersmoke** [comment]
@Gopalakrishnan11 I think I might see an issue with your rule.  The ordering of each part of a pattern is crucial.  

Also, if there are any tokens in a pattern which do not have the "OP" set to "?", then they must be present to have a match.  If you want a token be optionally present, check other context rules for examples of how to set the "OP" modifier to be "?" (i.e., optional, or 0 or 1 present to match)

I think you have most of the pattern right, but it doesn't match the order or tokens in your example sentence.

Your sentence says "medication not helps cure anxiety"

But your rule shows that the ordering withing the pattern is like this and also the word "to" which is not in your example sentence:
* ["not", "n't"]
* ["to"]
* ["cures","curing","cure"]
* ["help","helps"]

Does the rule do what you expect if you modified the rule like this to change the ordering and remove "to"??

* ["not", "n't"]
* ["help","helps"]
* ["cures","curing","cure"]
